ft-sdk/src/cookie.rs

```diff
@@ -1,3 +1,69 @@
+#[derive(Clone)]
+pub struct Cookie {
+    name: String,
+    value: String,
+    /// Indicates the path that must exist in the requested URL for the browser to send the Cookie
+    /// header.
+    path: Option<String>,
+    // TODO: add support for session cookies and expires:
+    // https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#removal_defining_the_lifetime_of_a_cookie
+    /// Indicates the number of seconds until the cookie expires. A zero or negative number will
+    /// expire the cookie immediately
+    max_age: Option<i64>,
+}
+
+impl Cookie {
+    /// Create a new cookie with the given name and value.
+    pub fn new(name: &str, value: &str) -> Self {
+        Self {
+            name: name.to_string(),
+            value: value.to_string(),
+            max_age: None,
+            path: None,
+        }
+    }
+
+    /// Set the max-age (in seconds) attribute for the cookie
+    ///
+    /// Panics if `max_age` is greater than 400 days (34560000 seconds)
+    /// See:
+    /// https://httpwg.org/http-extensions/draft-ietf-httpbis-rfc6265bis.html#name-the-max-age-attribute
+    pub fn set_max_age(&mut self, max_age: i64) {
+        assert!(max_age <= 34560000, "max_age must be less than 400 days");
```

Comment:
Maybe instead of panicking here we set the max_age to the max recommended value, and issue a warning that we have done so.